src/pages/index.js

```diff
@@ -144,3 +190,72 @@ const cardUrlInput = addCardFormElement.querySelector("#add-url");
 // Adding a New Card //
 
 addNewCardButton.addEventListener("click", () => addCardWithPopupForm.open());
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "89dc4b2f-fab0-42f3-ad8c-2593f7f5189c",
+    "Content-Type": "application/json",
+  },
+});
+
+function handleLikeCard(card, cardId, isLiked) {
+  api
+    .updateLikeStatus(cardId, isLiked)
+    .then((updatedCard) => {
+      card.setIsLiked(updatedCard.isLiked);
+    })
+    .catch((err) => console.error(err));
+}
+
+function renderCardsAfterUserInfo() {
+  return Promise.all([api.getInitialCards(), api.getUserInfo()]).then(
+    ([cards, userData]) => {
+      cardSection.renderItems(cards);
+      userInfo.setUserInfo({
+        name: userData.name,
+        job: userData.about,
+      });
+      userInfo.setUserAvatar({
+        avatar: userData.avatar,
+      });
+    }
+  );
+}
+
+renderCardsAfterUserInfo();
+
+const handleAvatarModal = new PopupWithForm("#avatar-modal", (data) => {
+  handleAvatarModal.renderModalFormLoading(true);
+  api
+    .editAvatar({
+      avatar: data.avatar,
+    })
+    .then((updatedAvatarInfo) => {
+      userInfo.setUserAvatar(updatedAvatarInfo);
+      handleAvatarModal.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => handleAvatarModal.renderModalFormLoading(false));
+});
+
+const avatarEditButton = document.querySelector(".avatar__edit-icon");
+avatarEditButton.addEventListener("click", () => {
+  handleAvatarModal.open();
+  handleAvatarModal.setEventListeners();
```

Comment:
`  handleAvatarModal.setEventListeners();` should be in line 247


`.setEventListeners()` should be removed from `opening` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This may cause a **`memory leak`**